src/utils/date-time-formatter.js

```diff
@@ -699,17 +703,27 @@ DateTimeFormatter.prototype.extractTokenValue = function (dateTimeStamp) {
  * @return {Date} : Native JS Date
  */
 DateTimeFormatter.prototype.getNativeDate = function (dateTimeStamp) {
-    if (dateTimeStamp instanceof Date) {
-        return dateTimeStamp;
-    } else if (isFinite(dateTimeStamp) && !!this.format) {
-        return new Date(dateTimeStamp);
+    let date = null;
+    if (Number.isFinite(dateTimeStamp)) {
+        date = new Date(dateTimeStamp);
+    } else if (!this.format && Date.parse(dateTimeStamp)) {
+        date = new Date(dateTimeStamp);
     }
+    else {
+        const dtParams = this.dtParams = this.parse(dateTimeStamp);
+        let dateParms = [];
+        if (dtParams.length) {
+            this.checkIfOnlyYear(dtParams.length) ? dateParms.unshift(null, dtParams[0], 0, 1)
```

Comment:
Shouldn't this be done by the parser method?

Reply:
done